geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufOpsProcessor.java

```diff
@@ -86,6 +85,9 @@ private Result processOperation(ClientProtocol.Message request, MessageExecution
       logger.error(exception);
       return Failure.of(BasicTypes.ErrorCode.INVALID_REQUEST,
           "Invalid execution context found for operation.");
+    } catch (UnsupportedOperationException ex) {
+      return Failure.of(BasicTypes.ErrorCode.UNSUPPORTED_OPERATION,
+          "Unsupported operation. This may be due to a region type that does not support that operation.");
```

Comment:
Can we return the message of the underlying exception to the user, maybe? I suspect a lot of UnsupportedOperationExceptions are not thrown from Region. Also, we should still log the stack trace I think. Otherwise we may never be able to track down what method actually threw UnsupportedOperation.